lsp-types/src/Language/LSP/Types/Uri.hs

```diff
@@ -151,37 +184,75 @@ platformAdjustToUriPath systemOS srcPath
           FPP.addTrailingPathSeparator (init drv)
       | otherwise = drv
 
+#ifdef OS_PATH
```

Comment:
I suggest isolating all the CPP and filepath related stuff in an internal `System.OsPath.Compat` module or something.

Reply:
What about `Language.LSP.Types.Uri.OsPathCompat`? Not choosing a general name like `System.OsPath.Compat` is because the CPP stuff introduced is probably only suitable for `NormalizedFilePath`.

Most other modules should simply switch between `FilePath` and `OsPath`. A type alias and a conditional import should suffice. Let's worry about that when ghc 9.6 is released.

The main distinction is that most modules only need to care about the "correctness" `OsPath` provides, not the performance gain.

Reply:
Sure, although we probably don't have to worry too much about the name if we don't export the module.